app/models/authenticator/httpd.rb

```diff
@@ -4,15 +4,25 @@ def self.proper_name
       'External httpd'
     end
 
-    def authorize_queue(username, request)
-      user_attrs = {:username  => username,
-                    :fullname  => request.headers['X-REMOTE-USER-FULLNAME'],
-                    :firstname => request.headers['X-REMOTE-USER-FIRSTNAME'],
-                    :lastname  => request.headers['X-REMOTE-USER-LASTNAME'],
-                    :email     => request.headers['X-REMOTE-USER-EMAIL']}
-      membership_list = (request.headers['X-REMOTE-USER-GROUPS'] || '').split(/[;:]/)
+    def authorize_queue(username, request, options, *_args)
+      if options[:authorize_only] == true
+        ext_user_attrs = get_user_attrs(username)
+        user_attrs = {:username  => username,
+                      :fullname  => ext_user_attrs["displayname"],
+                      :firstname => ext_user_attrs["givenname"],
+                      :lastname  => ext_user_attrs["sn"],
+                      :email     => ext_user_attrs["mail"]}
+        membership_list = MiqGroup.get_httpd_groups_by_user(username)
+      else
+        user_attrs = {:username  => username,
+                      :fullname  => request.headers['X-REMOTE-USER-FULLNAME'],
+                      :firstname => request.headers['X-REMOTE-USER-FIRSTNAME'],
+                      :lastname  => request.headers['X-REMOTE-USER-LASTNAME'],
+                      :email     => request.headers['X-REMOTE-USER-EMAIL']}
+        membership_list = (request.headers['X-REMOTE-USER-GROUPS'] || '').split(/[;:]/)
+      end
```

Comment:
This looks like it could be DRY-ed up a little or perhaps create support methods. I suggest considering something like this:

```ruby
def get_ext_user_attrs(username)
    ext_user_attrs = get_user_attrs(username)
    user_attrs = {:username  => username,
                  :fullname  => ext_user_attrs["displayname"],
                  :firstname => ext_user_attrs["givenname"],
                  :lastname  => ext_user_attrs["sn"],
                  :email     => ext_user_attrs["mail"]}

    [user_attrs, MiqGroup.get_httpd_groups_by_user(username)]
end

def get_user_attrs_from_headers(username, request)
    user_attrs = {:username  => username,
                  :fullname  => request.headers['X-REMOTE-USER-FULLNAME'],
                  :firstname => request.headers['X-REMOTE-USER-FIRSTNAME'],
                  :lastname  => request.headers['X-REMOTE-USER-LASTNAME'],
                  :email     => request.headers['X-REMOTE-USER-EMAIL']}

    [user_attrs, (request.headers['X-REMOTE-USER-GROUPS'] || '').split(/[;:]/)]
end

def authorize_queue(username, request, options, *_args)
  if options[:authorize_only] == true
    user_attrs, membership_list = get_ext_user_attrs(username)
  else
    user_attrs, membership_list = get_user_attrs_from_headers(username, request)
  end 

  super(username, request, {}, user_attrs, membership_list)
end 

```